### ISSUE TITLE
Expose available variants in fonts API endpoint

### ISSUE THREAD

**Issue by jauyong** [GoogleForCreators/web-stories-wp#876]
## Feature Description

We are using `web-stories/v1/fonts` api to fetch the fonts from the wordpress. Currently there is no information for variants of individual fonts. We need to add variants like `italilc` to that api to know if the font has italic font as well.

See #670

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance Criteria
1. Add variants fields to font fetch api.

**Comment by jauyong** [comment]
@ndev1991 can you please add details on what is required to complete #670 and reassign when done?

**Comment by ndev1991** [comment]
@spacedmonkey @miina can we have this ticket as a priority ? Thank you
(see [Roboto](https://fonts.google.com/specimen/Roboto) as example)

**Comment by spacedmonkey** [comment]
@ndev1991 Until you provide details on what extra fields you need in the API, neither of us can start work on this ticket. Please fill out detail and I will take a look.

**Comment by ndev1991** [comment]
@spacedmonkey Here is what I understand for this one.
`const fontStylesheet = document.createElement('link');`
      `fontStylesheet.id = id;`
     ` fontStylesheet.href = src;`
      `fontStylesheet.rel = 'stylesheet';`
      `fontStylesheet.type = 'text/css';`
      `fontStylesheet.media = 'all';`
      `fontStylesheet.crossOrigin = 'anonymous';`
     ` document.head.appendChild(fontStylesheet);`

We have this part to insert required font when selected. And currently the `fonts` api return src with that `fontWeights` included. But not the variants. Like the example here
https://fonts.google.com/specimen/Roboto?selection.family=Roboto:ital,wght@0,100;0,400;0,700;1,100;1,400;1,700&sidebar.open

So I think we need to include the variants with correct src that will include correct fonts to this document.
@pbakaus Would you confirm is this correct understanding?

**Comment by ndev1991** [comment]
> @spacedmonkey Here is what I understand for this one.
> `const fontStylesheet = document.createElement('link');`
> `fontStylesheet.id = id;`
> `fontStylesheet.href = src;`
> `fontStylesheet.rel = 'stylesheet';`
> `fontStylesheet.type = 'text/css';`
> `fontStylesheet.media = 'all';`
> `fontStylesheet.crossOrigin = 'anonymous';
> 
> ```
>   document.head.appendChild(fontStylesheet);`
> ```
> 
> We have this part to insert required font when selected. And currently the `fonts` api return src with that `fontWeights` included. But not the variants. Like the example here
> https://fonts.google.com/specimen/Roboto?selection.family=Roboto:ital,wght@0,100;0,400;0,700;1,100;1,400;1,700&sidebar.open
> 
> So I think we need to include the variants with correct src that will include correct fonts to this document.
> @pbakaus Would you confirm is this correct understanding?

**Comment by ndev1991** [comment]
> > @spacedmonkey Here is what I understand for this one.
```
> > `const fontStylesheet = document.createElement('link');`
> > `fontStylesheet.id = id;`
> > `fontStylesheet.href = src;`
> > `fontStylesheet.rel = 'stylesheet';`
> > `fontStylesheet.type = 'text/css';`
> > `fontStylesheet.media = 'all';`
> > `fontStylesheet.crossOrigin = 'anonymous';
> >   document.head.appendChild(fontStylesheet);`
> > ```
> > 
> > 
> > We have this part to insert required font when selected. And currently the `fonts` api return src with that `fontWeights` included. But not the variants. Like the example here
> > https://fonts.google.com/specimen/Roboto?selection.family=Roboto:ital,wght@0,100;0,400;0,700;1,100;1,400;1,700&sidebar.open
> > So I think we need to include the variants with correct src that will include correct fonts to this document.
> > @pbakaus Would you confirm is this correct understanding?

**Comment by ndev1991** [comment]
@miina I think you worked on the textStyle panel and the textElement. Can you help to clarify this issue?

**Comment by spacedmonkey** [comment]
I am not sure what is the issue is here. When I picked raleway, this is the requested font file. 
https://fonts.googleapis.com/css?family=Raleway%3A100%2C100i%2C200%2C200i%2C300%2C300i%2C400%2C400i%2C500%2C500i%2C600%2C600i%2C700%2C700i%2C800%2C800i%2C900%2C900i&subset=latin%2Clatin-ext&display=swap

As you can see from this request, we are loading italic variant and many different weights. So what is issue here?

**Comment by swissspidy** [comment]
To me that sounds like we are requesting too much.

By default only the regular font weight should be loaded.

In the REST API endpoint we'd need to expose which variants the font supports to show the appropriate options in the dropdown.

Then, when actually using italic or bold (or bolid+italic), we load those too.

**Comment by spacedmonkey** [comment]
@swissspidy Is this really required at this stage? Feel like a bug for post alpha. This isn't a bug, it is an optimisation.

**Comment by miina** [comment]
@ndev1991 Looks like the italic is already loaded with the font file, #670 is unblocked and doesn't need API updates.

I suppose we can close this issue and create a follow-up issue for optimization?

**Comment by ndev1991** [comment]
> @ndev1991 Looks like the italic is already loaded with the font file, #670 is unblocked and doesn't need API updates.
> 
> I suppose we can close this issue and create a follow-up issue for optimization?

@miina Seems we have all available fonts loading on src. And my question here is how to detect if the font has italic variant available. And as mentioned on issue #670 current italic toggle uses default browser italic. And if we need to use font italic we need to know if the font has italic variants?

**Comment by miina** [comment]
> @miina Seems we have all available fonts loading on src. And my question here is how to detect if the font has italic variant available. And as mentioned on issue #670 current italic toggle uses default browser italic. And if we need to use font italic we need to know if the font has italic variants?

Aren't these automatically used if available? I guess `bold` is the one that might need some reworking now since it needs to be in sync with which font-weight is chosen (it's currently just wrapping with `strong` tags.

**Comment by ndev1991** [comment]
> > @miina Seems we have all available fonts loading on src. And my question here is how to detect if the font has italic variant available. And as mentioned on issue #670 current italic toggle uses default browser italic. And if we need to use font italic we need to know if the font has italic variants?
> 
> Aren't these automatically used if available? I guess `bold` is the one that might need some reworking now since it needs to be in sync with which font-weight is chosen (it's currently just wrapping with `strong` tags.

hmm... I see. Then I think we don't need issue #670 anymore? @miina @jauyong

**Comment by miina** [comment]
Since #670 was related to variants generally then it could probably stay to address only the `bold` (and extra bold, etc.), the title of the issue would need an update -- however, AFAIK it's not a critical priority at this point.

cc @pbakaus

**Comment by jauyong** [comment]
Removing this from the board for now and treating as a non alpha required issue.
